data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -102,9 +104,22 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
 
         includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
         excludeKeysSet.addAll(keyValueProcessorConfig.getExcludeKeys());
+        defaultKeysMap.putAll(keyValueProcessorConfig.getDefaultKeys());
+        if (!defaultKeysMap.isEmpty()) {
+            defaultKeysSet.addAll(defaultKeysMap.keySet());
+        }
 
-        validateKeySets(includeKeysSet, excludeKeysSet);
+        validateKeySets(includeKeysSet, excludeKeysSet, defaultKeysSet);
 
+        final Set<String> includeDefaultCheckSet = new HashSet<String>(defaultKeysSet);
+        includeDefaultCheckSet.retainAll(includeKeysSet);
+        if (!includeDefaultCheckSet.isEmpty()) {
+            for (final String overlap_key : includeDefaultCheckSet) {
+                defaultKeysMap.remove(overlap_key);
```

Comment:
If the overlap_key does not appear in the message, this will not yield the expected result. e.g.

message = "key2=value2"
include_keys = [key1]
default_keys = {"key1": "value1"}
expected output
```
{
"key1": "value1"
}
```
but since you removed overlap_key from the defaultKeysMap key1 will not be imputed with value1 in the result event.

Reply:
Please also add the above example into test case.